### PR TITLE
Expose BatchEntryDelete On CLI

### PR DESCRIPTION
**Pull request by JU4N98**
**Affected functionality**
This PR affects the entry delete on cli.

**Description of change**
This PR allows:
* The use of entryID flag multiple times (like selector flag of entry create).
* The use of the new flag data, which specifies the path to a JSON with the entryIDs to be deleted or just "-" in case that the JSON should be taken from stdin. Follows an example of this JSON:
```
{
    Ids : ["entry-0", "entry-1"]
}
```
In both cases, the output is similar to entry create, first the succeeded entry deletions are enumerated and then the failed ones.

**Which issue this PR fixes**
This PR fixes the[ issue #4248](https://github.com/spiffe/spire/issues/4248).

### CODE REVIEW

File: cmd/spire-server/cli/entry/delete.go [hunk 1]
```diff
@@ -26,9 +28,10 @@ func newDeleteCommand(env *commoncli.Env) cli.Command {
 
 type deleteCommand struct {
 	// ID of the record to delete
-	entryID string
-	env     *commoncli.Env
-	printer cliprinter.Printer
+	entriesIDs StringsFlag
```
Comment (evan2645) [inline cmd/spire-server/cli/entry/delete.go:31]: nit, this should probably be
```suggestion
	entryIDs StringsFlag
```

File: cmd/spire-server/cli/entry/delete.go [hunk 2]
```diff
@@ -40,16 +43,54 @@ func (*deleteCommand) Synopsis() string {
 }
 
 func (c *deleteCommand) AppendFlags(f *flag.FlagSet) {
-	f.StringVar(&c.entryID, "entryID", "", "The Registration Entry ID of the record to delete")
+	f.Var(&c.entriesIDs, "entryID", "The Registration Entry ID of the record to delete. Can be used more than once")
+	f.StringVar(&c.path, "data", "", "Path to a file containing deletion JSON (optional). If set to '-', read the JSON from stdin.")
```
[the remark this answers is not in the document]
Comment (evan2645) [reply cmd/spire-server/cli/entry/delete.go:47]: I think we should mention something about batch deletion here ... maybe like:
```suggestion
	f.StringVar(&c.path, "data", "", "Path to a file containing a JSON structure for batch deletion (optional). If set to '-', read from stdin.")
```

File: cmd/spire-server/cli/entry/delete.go [hunk 3]
```diff
@@ -73,12 +118,30 @@ func (c *deleteCommand) prettyPrintDelete(env *commoncli.Env, results ...interfa
...
+			failed = append(failed, result)
+		}
 	}
+
+	for _, result := range succeeded {
+		env.Printf("Deleted entry with ID: %s\n", result.Id)
+	}
+
+	for _, result := range failed {
+		env.ErrPrintf("Failed to delete entry with ID %s (code: %s, msg: %q)\n",
```
Comment (evan2645) [inline cmd/spire-server/cli/entry/delete.go:136]: We should probably print out a newline or two between the success and error lines

File: cmd/spire-server/cli/entry/delete.go [hunk 4]
```diff
@@ -73,12 +118,30 @@ func (c *deleteCommand) prettyPrintDelete(env *commoncli.Env, results ...interfa
...
+			result.Id,
+			codes.Code(result.Status.Code),
+			result.Status.Message)
+	}
+
+	if len(failed) > 0 {
+		return errors.New("failed to delete one or more entries")
```
Comment (evan2645) [inline cmd/spire-server/cli/entry/delete.go:143]: The CLI command should only error if the call as a whole has errored

**Review by evan2645 — COMMENTED** [review]
Thank you very much for the contribution @JU4N98! And sorry for the delay on my part

Left a handful of comments, mostly around UX. It leads me to wonder about the output we print ... if folks are providing programmatic input, should we be providing programatic output? What is the path to recovering from a batch delete call where half the requests failed? just thinking out loud ...

File: cmd/spire-server/cli/entry/delete.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (JU4N98) [reply cmd/spire-server/cli/entry/delete.go:47]: Sure, i think that file is a better description than data.

File: cmd/spire-server/cli/entry/delete.go [hunk 5]
```diff
@@ -40,16 +43,54 @@ func (*deleteCommand) Synopsis() string {
 }
 
 func (c *deleteCommand) AppendFlags(f *flag.FlagSet) {
-	f.StringVar(&c.entryID, "entryID", "", "The Registration Entry ID of the record to delete")
+	f.Var(&c.entriesIDs, "entryID", "The Registration Entry ID of the record to delete. Can be used more than once")
```
[the remark this answers is not in the document]
Comment (JU4N98) [reply cmd/spire-server/cli/entry/delete.go:46]: Yes, you are right, i didn't notice that entry create doesn't support batch create without a file. I'm going to fix that and keep the file flag.

File: cmd/spire-server/cli/entry/delete.go (hunk 4 above)
Comment (JU4N98) [reply cmd/spire-server/cli/entry/delete.go:143]: I followed create.go as a guide:
https://github.com/spiffe/spire/blob/5aa2122fbcc892979b15be93a20ebbb6f2c41438/cmd/spire-server/cli/entry/create.go#L284

Maybe I misunderstood the code of entry create, but I think that we are doing the same there.

Thanks for the comments!

Comment (evan2645) [reply cmd/spire-server/cli/entry/delete.go:145]: Ah ok ... consistency is best 😊 thanks for pointing that out

**Review by evan2645 — APPROVED** [review]
Thank you @JU4N98 !!

**Comment by evan2645** [comment]
Hey @JU4N98, looks like the Windows unit tests need to be fixed up following changes to help text output: https://github.com/spiffe/spire/actions/runs/5907969226/job/16186854897?pr=4371

Mind patching it so we can get this merged in for [1.8.0](https://github.com/spiffe/spire/milestone/30)?

**Comment by JU4N98** [comment]
> Hey @JU4N98, looks like the Windows unit tests need to be fixed up following changes to help text output: https://github.com/spiffe/spire/actions/runs/5907969226/job/16186854897?pr=4371
> 
> Mind patching it so we can get this merged in for [1.8.0](https://github.com/spiffe/spire/milestone/30)?

hey @evan2645 windows unit test are fixed right now

**Review by evan2645 — APPROVED** [review]

**Pull request merged**